src/app/parser/impl/LabradorParser.ts

```diff
@@ -0,0 +1,232 @@
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+import * as Cheerio from 'cheerio';
+import Article from 'base/Article';
+
+import GenericParser from './GenericParser';
+
+import * as app from 'app/util/applib';
+import {getContent, ProvidedContent} from 'app/parser/util/labradorparser/ProvidedContent';
+import ArticleItemType from 'base/ArticleItemType';
+import ArticleAuthor from 'base/ArticleAuthor';
+import ArticleItem from 'base/ArticleItem';
+
+const log = app.createLog();
+
+interface ParserWorkflow {
+	version : any;
+	authors : any;
+	url: any;
+	items : any;
+}
+
+export default class LabradorParser extends GenericParser {
+
+	protected select : Cheerio;
+	protected jsonArticle: any;
+	private resultData : ParserWorkflow;
+
+	/**
+	 * Derived method that is called by parser factory
+	 */
+	parse() : Promise <Article> {
+		return this.init()
+		.then(() => {
+			const providedContent = this.getProvidedContent();
+			const jsonContent = this.getParsedContent();
+
+			return this.finalizeAnswer(jsonContent, providedContent);
+		})
+		.catch((error) => {
+			return Promise.reject(error);
+		});
+	}
+
+	/**
+	 * Simple response validation is here.
+	 * Also sets jsonArticle property.
+	 */
+	init() : Promise <any> {
+		if (this.validateApiResponse(this.rawArticle)) {
+			this.jsonArticle = this.rawArticle.result[0];
+			return Promise.resolve();
+		}
+		const error = 'Bad source response';
+		log(error);
+		return Promise.reject(new Error(error));
+	}
+
+	/**
+	 * Validating response
+	 */
+	validateApiResponse(rawArticle): boolean {
+		if (!rawArticle.result || ! Array.isArray(rawArticle.result) ) {
+			return false;
+		}
+		if (rawArticle.result.length !== 1 || !rawArticle.result[0]) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * We still have html content that should be parsed as usual.
+	 */
+	getParsedContent() {
+		if (this.jsonArticle.bodytextHTML && this.jsonArticle.bodytextHTML) {
+			const $ = Cheerio.load(this.jsonArticle.bodytextHTML);
+			const items = [];
+			$('*').each((i, elem) => {
+				let item;
+				switch (elem.name) {
+					case 'h2':
+					case 'h3':
+					case 'h4':
+					case 'h5':
+					case 'h6':
+						item = this.createSubHeadingEl($(elem).text());
+						break;
+					case 'a':
+						item = this.createLinkEl(elem.attribs.href, $(elem).text());
+						break;
+					case 'body':
+						break;
+					case 'head':
+						break;
+					case 'html':
+						break;
+					case 'p':
+					default:
+						item = this.createParagraphEl($(elem).text());
+						break;
+				}
+				if (item !== undefined) {
+					items.push(item);
+				}
+			});
+			return items;
```

Comment:
Only return items in the end. If they are not set - it is empty array. No unusefull returns